File: WEEK3/Day1/bubbles/bubble-sort.py
sequence = [4, 3, 5, 0, 1]
swaps = 0

import random

arr = [random.randint(1, 10) for _ in range(10)]

def bubble_swap(arr):
    print(f'Initial array: {arr}')
    swaps = 0

    for i in range(len(arr)):
        if arr[i] < arr[i - 1]:
            arr[i], arr[i-1] = arr[i-1], arr[i]
            swaps += 1
    print(f'{swaps} swaps')
    print(f'result: {arr}')
# bubble_swap(arr)
        
    

def big_bubble_swap(arr):
    print(f'Initial array: {arr}')
    swaps = 0
   
    

    for _ in range(len(arr)):
        made_swap = False
        for i in range(1, len(arr)):
            if arr[i] < arr[i-1]:
                arr[i], arr[i - 1] = arr[i - 1], arr[i]
                swaps += 1
                made_swap = True
        if not made_swap:
            break
      
   

   
    print(f"Sorted array: {arr}")
    print(f"Swaps: {swaps}")

big_bubble_swap(arr)

def sorting(arr):
    print(f"Initial array: {arr}")
    
    sorted_arr = sorted(arr)
    print(f"Sorted array using 'sorted()': {sorted_arr}")
    print(f"original array remains unchanged: {arr}")

# sorting(arr)

File: WEEK3/Day1/bubbles/bubbleSort.js
// var sequence = [4, 3, 5, 0, 1]
// var swaps = 0

const arrayLength = 10;
const maxRandomValue = 10

const randomArray = Array.from({length: arrayLength }, () => Math.floor(Math.random() * maxRandomValue) + 1);

console.log(randomArray)

const bubblesort = (arr) => {
    let n = arr.length;
    let swaps = 0;

    for (let i = 0; i < n - 1; i++) {
        if (arr[i] > arr[i + 1]) {
            [arr[i], arr[i +1]] = [arr[i + 1], arr[i]]
            swaps += 1
        } 
    }
    console.log("swaps = " + swaps)
    return arr

}
const blowingBubbles = bubblesort(randomArray)
console.log("blowing bubbles: " + blowingBubbles)

// const bigBubbleSort = (arr) => {
//     let n = arr.length;
//     let swapped;
//     let swaps = 0;
//     do {
//         swapped = false;
//         for (let i = 0; i < n - 1; i++) {
//             if (arr[i] > arr[i + 1]) {
//                 [arr[i], arr[i + 1]] = [arr[i + 1], arr[i]];
//                 swapped = true;
//                 swaps += 1
//             }
//         } 
//         n--;
//     } while (swapped);
//     console.log("Swaps: " + swaps)
//     return arr;

// }

// const sortedArr = bigBubbleSort(randomArray);
// console.log("Sorted Array:", sortedArr);


